scripts/combine-crossreg.R

```diff
@@ -0,0 +1,237 @@
+# Now add matches made from the title matching algorithm. Will follow similar logic to steps above
+
+# Modify structure of title match table to make it easier to integrate with larger match table
+trn_trn_titles <- title_matches |>
+  filter(id != euctr_id) |>   # filter out self-references
+  select(trn1 = id, trn2 = euctr_id, is_title_matched = has_crossreg_eudract_tm)
+
+# Integrate title matched TRNs
+trn_trn_titles_tidy <- trn_trn_results_tidy |>
+  mutate(is_title_matched = NA) |>
+  rows_upsert(trn_trn_titles, by = c("trn1", "trn2"))
+
+####################################################################################################################
+## Finally add matches from TRNs found in publications
+
+# Unnest TRNs in pubs table
+trn_trn_pubs_long <- publications |>
+  rename(trn1 = primary_IV_id) |>
+  mutate(trn_si_long = strsplit(trns_si, ";"),
+         trn_abs_long = strsplit(trns_abs, ";"),
+         trn_ft_long = strsplit(trns_ft, ";")) |> # Split trns_si by semicolon to get individual trial IDs
+  unnest_longer(trn_si_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_si_long != trn1 | is.na(trn_si_long)) |> # filter out self-references
+  unnest_longer(trn_abs_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_abs_long != trn1 | is.na(trn_abs_long)) |> # filter out self-references
+  unnest_longer(trn_ft_long) |> # unnest longer to allow pairwise comparisons
+  filter(trn_ft_long != trn1 | is.na(trn_ft_long)) |> # filter out self-references
+  select(trn1, contains("long"))
+
+# Identify unique TRN pairings from pubs fields, fill out boolean information to identify pub source of match
+trn_trn_pubs <- trn_trn_pubs_long |>
+  rowwise() |>
+  mutate(trn2 = list(unique(c(trn_si_long, trn_abs_long, trn_ft_long)) |> na.omit())) |>
+  unnest_longer(trn2) |>
+  group_by(trn1, trn2) |>
+  summarise(trn2_in_pub_si = if_else(any(trn2 == trn_si_long), TRUE, FALSE),
+            trn2_in_pub_abs = if_else(any(trn2 == trn_abs_long), TRUE, FALSE),
+            trn2_in_pub_ft = if_else(any(trn2 == trn_ft_long), TRUE, FALSE)) |>
+  rowwise() |>
+  mutate(trn1inreg2 = trn1 %in% trn_registries$trn2[trn_registries$trn1 == trn2],
+         trn2inreg1 = trn2 %in% trn_registries$trn2[trn_registries$trn1 == trn1]) |>
+  select(trn1, trn2, trn1inreg2, trn2inreg1, contains("pub")) |>
+  ungroup()
+
+# Integrate pub matches to larger match table
+trn_trn_pubs_tidy <- trn_trn_titles_tidy |>
+  mutate(trn2_in_pub_si = NA, trn2_in_pub_abs = NA, trn2_in_pub_ft = NA) |>
+  rows_upsert(trn_trn_pubs, by = c("trn1", "trn2")) |>
+  rowwise() |>
+  mutate(registry1 = case_when(is.na(registry1) ~ which_registry(trn1),
+                               .default = registry1),
+         registry2 = case_when(is.na(registry2) ~ which_registry(trn2),
+                               .default = registry2)) |>
+  ungroup() |>
```

Comment:
this step is indeed necessary, but here you're not just filling in the blanks for `registry1` and `registry2` for matches based on publications. You also have blanks in these variables from when you brought in matches based on protocol/results sponsor code and based on title matching. This code is ok but could be its own separate chunk to indicate that this is where any blanks in these variables are being filled (not just from publications processing)

Reply:
@delwen, @lucfernandez: I'm not able to completely chime in on this, as I don't have the data to run this script, but conceptually, I understood `registry1` and `registry2` to mean whether any data from 1 or 2 points to the other, hence trn, codes, title, etc. If I understand @delwen's comment, this is what you mean too

Reply:
no I think we have a misunderstanding here. @maia-sh what I think you're referring to are the variables `trn1inreg2` and `trn2inreg1` (which btw should just be for whether TRN in registry), whereas `registry1` and `registry2` is just encoding the name of the registries associated with a potential cross-registration.

My comment here is just about whether this chunk should be where publication data is being processed, or rather its own chunk, since it's filling in missing registry names for matches based on publications but also protocol/results sponsor codes and title matching.

Reply:
yes I think a separate step would be clearer!